### PR TITLE
EE-1095: Auction: consolidate constants, type aliases, and helpers

### PR DESCRIPTION
**Pull request by henrytill**
Ref: https://casperlabs.atlassian.net/browse/EE-1095

### CODE REVIEW

File: types/src/auction/detail.rs [hunk 1]
```diff
@@ -1,17 +1,169 @@
...
+    P: StorageProvider + RuntimeProvider + ?Sized,
+    T: ToBytes + CLTyped,
+{
+    let key = provider.get_key(name).ok_or(Error::MissingKey)?;
+    let uref = key.into_uref().ok_or(Error::InvalidKeyVariant)?;
+    provider.write(uref, value)?;
+    Ok(())
+}
+
+pub fn get_bids<P>(provider: &mut P) -> Result<Bids>
```
Comment (Fraser999) [inline types/src/auction/detail.rs:39]: I don't this was done in this PR, so probably out of scope, but I think we should prefer to not prefix getters with `get_`.

**Review by Fraser999 — APPROVED** [review]
One nitpick which is probably out of scope for this PR, but otherwise lgtm.

File: types/src/auction/detail.rs (hunk 1 above)
Comment (henrytill) [reply types/src/auction/detail.rs:39]: Much agreed

**Pull request merged**